### ISSUE TITLE
Browser debuggers do not work correctly with Marionette using source map

### ISSUE THREAD

**Issue by blikblum** [marionettejs/backbone.marionette#3328]
### Description

 1. Trying to debug Marionette code in Chrome or Firefox debugger do not work when using source map

### Expected behavior

Debugger should work with source map

### Actual behavior

Without source map, debugging works fine

When using source map the place where breakpoint is set and the highlighted line differs from the actual code being run. As example load a simple marionette app like http://codepen.io/blikblum/pen/QpQxZb (to avoid confusion factor from CodePen shell, i extracted the code from the debug view into a single HTML file and ran locally with a fresh build from next branch).

In chrome debugger, put a break point at src/region.js line 202 (`this._restoreEl()`). Try to step inside function. It will go inside `Events.on` method instead of going into `_restoreEl`.

Now put the breakpoint at line 204, over `view.on('before:destroy', this._restoreEl, this)`. Look at elements tab. It should have 'div#hook' element. Try to step over `view.on` call. Look at the elements: 'div#hook' is now replaced by 'div "Foo"' meaning that `this.replaceEl` call is already done although, by the line being highlighted it should be called only after the next step over.

### Environment

1. Marionette version: 3.2 (next)
2. Backbone version: 1.3.3

**Comment by paulfalgout** [comment]
Hmm.. I'm not sure what the issue is.  Comparing the sourcemaps to the code it looks like it links up fine.. try http://sokra.github.io/source-map-visualization/  But there's so many sourcemap issues out there right now, I'm not surprised.  I'm not sure what we'll be able to do to fix it either.

**Comment by paulfalgout** [comment]
I recently tried @jasonLaster 's https://github.com/devtools-html/debugger.html which seemed to do better than chrome devtools.  It didn't quite work either (probably not it's fault) but it did seem to be less broken feeling.. essentially it never made a breakpoint to the wrong place like in chrome, but it wouldn't let me break to particular spots.  I still need to do more digging.

**Comment by blikblum** [comment]
Related: https://github.com/rollup/rollup/issues/1370

We should track the evolution of the issue so a possibility is to use the source map generated by rollup. Up now, using rollup to produce the source map exhibits the same behavior

**Comment by paulfalgout** [comment]
I _think_ this is working in v4.  Seems to be.  Don't know what changed or fixed it, but I'm going to close this unless a problem is discovered.  But I was able to update the codepen and go to the line without issue.
